ee/src/main/java/org/jboss/as/ee/metadata/AbstractEEAnnotationProcessor.java

```diff
@@ -58,6 +58,12 @@ public final void deploy(final DeploymentPhaseContext phaseContext) throws Deplo
             for (Map.Entry<String, ClassAnnotationInformation<?, ?>> entry : data.entrySet()) {
                 EEModuleClassDescription clazz = eeModuleDescription.addOrGetLocalClassDescription(entry.getKey());
                 clazz.addAnnotationInformation(entry.getValue());
+
+                if(entry.getValue()!=null && entry.getValue().getAnnotationType()!=null && entry.getValue().getAnnotationType().toString().contains("Startup") && deploymentUnit.getParent()!=null) {
+                    List<String> startupBeanModules = deploymentUnit.getParent().getAttachmentList(org.jboss.as.ee.component.Attachments.DEPLOYMENT_STARTUP_MODULES);
+                    if(startupBeanModules!=null && !startupBeanModules.contains(deploymentUnit.getName()))
+                        deploymentUnit.getParent().addToAttachmentList(Attachments.DEPLOYMENT_STARTUP_MODULES, deploymentUnit.getName());
+                }
             }
```

Comment:
`getAttachmentList` returns non-null, so no need for null check for `startupBeanModules`